firmware/Core/Src/telecommands/telecommand_parser.c

```diff
@@ -153,6 +155,60 @@ uint8_t TCMD_get_suffix_tag_uint64(const char *str, const char *tag_name, uint64
     return 0;
 }
 
+/// @brief Searches for a `str` like `\@tag_name=xxxx`, and sets uint8_t array `xxxx` into `value_dest`.
+/// @param str The string (haystack) to search with for the tag.
+/// @param tag_name The tag to search for, including the '@' and '='.
+/// @param value_dest The destination for the value. `*value_dest` will be set.
+/// @return 0 if the tag was found successfully. >0 if the tag was not found, or there was an error.
+/// @note This function will return an error if the character after the string is not in: `#\@\0!`
+uint8_t TCMD_get_suffix_tag_hex_array(const char *str, const char *tag_name, uint8_t *value_dest) {
+    // Find the tag in the string
+    int16_t tag_index = GEN_get_index_of_substring_in_array(str, strlen(str), tag_name);
+    if (tag_index < 0) {
+        return 1;
+    }
+
+    // Find the start of the value, then do safety check.
+    uint16_t value_start_index = tag_index + strlen(tag_name);
```

Comment:
Please use const for all variables in this scope which are assigned to only once